File: src/main/java/core/procedures/io/CurrentInputPort.java
package core.procedures.io;

import core.Repl;
import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class CurrentInputPort extends AFn {

  public CurrentInputPort() {
    super(new FnArgsBuilder().max(0).build());
  }

  @Override
  public String getName() {
    return "current-input-port";
  }

  @Override
  public Object apply0() {
    return Repl.getCurrentInputPort();
  }
}

File: src/main/java/core/procedures/io/CurrentInputPort.kt
package core.procedures.io

import core.Repl
import core.procedures.AFn
import core.procedures.FnArgsBuilder

class CurrentInputPort : AFn(FnArgsBuilder().max(0).build()) {

    override val name: String
        get() = "current-input-port"

    override fun apply0(): Any {
        return Repl.currentInputPort
    }
}